### ISSUE TITLE
Are the test set and the validation set the same?

### ISSUE THREAD

**Issue by JiuZeongit** [C19h/EGST#1]
Hi，Thank you for your wonderful and enlightening work! 

When I reproduce this repo, I found that the test set and the validation share the path. In graphtransformer_dataset.py, we can see this repo inputs val_dataloader with the path graph_test_dir, which points to the test file in Day dataset. when run day_eva_model.py, it inputs the path graph_test_dir, too.

**Comment by C19h** [comment]
Thank you very much for your question. In the gait recognition dataset, there is no difference between the test and validation sets. In order to check the performance of each epoch during training, or to evaluate the trained model, the test set is directly loaded into val_dataloader for convenience. So, here you can just think of test_dataloader as val_dataloader.

**Comment by JiuZeongit** [comment]
Thank you!

**Comment by wangtianye777-a11y** [comment]
Hello! Thank you for your innovative work and for sharing it! 
I am a student, and I was very impressed by the ideas and model framework presented in your EGST paper, so I am trying to reproduce your experimental results. However, I have encountered some problems. During training on the DVS128-Gait-Day dataset, I found that the validation accuracy after training only reached around 93%. For the CASIA_B dataset, the accuracy in step 1 was around 96%, but in step 2, it was only around 60%. Even after adjusting the parameters, I couldn't achieve the results presented in the paper, which is quite frustrating. Could you please provide a more complete code or suggest some methods to improve my results? 
I look forward to your reply!

**Comment by C19h** [comment]
感谢您提出的问题。可能存在的原因如下：1.运行环境不同，请试试按照给出的环境运行代码；2.请确保进行了预处理，及其相关超参数的设定；3.请检查是否意外改变了模型参数；4.在graphtransformer_dataset.py中不使用collate_sequences方法即不使用Batch类，而是把多个子graph视为一个列表，然后在模型中使用循环执行
---- Replied Message ----
***@***.***>Date10/01/2025 10:14 ***@***.***> ***@***.***>,
***@***.***>SubjectRe: [C19h/EGST] Are the test set and the validation set the same? (Issue #1)
wangtianye777-a11y left a comment (C19h/EGST#1)
Hello! Thank you for your innovative work and for sharing it!
I am a student, and I was very impressed by the ideas and model framework presented in your EGST paper, so I am trying to reproduce your experimental results. However, I have encountered some problems. During training on the DVS128-Gait-Day dataset, I found that the validation accuracy after training only reached around 93%. For the CASIA_B dataset, the accuracy in step 1 was around 96%, but in step 2, it was only around 60%. Even after adjusting the parameters, I couldn't achieve the results presented in the paper, which is quite frustrating. Could you please provide a more complete code or suggest some methods to improve my results?
I look forward to your reply!
